pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -295,68 +307,72 @@ func (pc *PodController) configureSriovAsSecondaryInterface(pod *corev1.Pod, net
 }
 
 // Configure Secondary Network Interface.
-func (pc *PodController) configureSecondaryInterface(pod *corev1.Pod, netinfo *SecondaryNetworkObject, podCNIInfo *cnipodcache.CNIConfigInfo, cniconfig []byte) error {
-	var err error
-	var ipamerr error
-	var ipamResult *current.Result
-	var cmdArgs *invoke.Args
+func (pc *PodController) configureSecondaryInterface(pod *corev1.Pod, network *netdefv1.NetworkSelectionElement, podCNIInfo *cnipodcache.CNIConfigInfo, cniConfig []byte) error {
 	// Generate and assign new interface name, If secondary interface name was not provided in Pod annotation.
-	if len(netinfo.InterfaceName) == 0 {
-		netinfo.InterfaceName = generatePodSecondaryIfaceName(podCNIInfo)
+	if len(network.InterfaceRequest) == 0 {
+		network.InterfaceRequest = generatePodSecondaryIfaceName(podCNIInfo)
 	}
-	if netinfo.InterfaceType == sriovInterfaceType {
```

Comment:
@antoninbas To clarify on the interface type at Annotation, The reason we kept interface type as part of Pod Annotation is to make it flexible at the Pod/CNF level to decide which interface they wanted to use based on the traffic needs/usage (for example, CNF with higher throughput needs can use SRIOV VFs).
As per the PR change, we moved it to the CR instance. Which means, any CNF pod which is getting configured to be part of this network will use SRIOV. for example, if the CNF gets scheduled in a server with no SRIOV NIC support (though its unlikely scenario in DC env?) secondary network config will fail (in current implementation, if the interface type is not provided, it will fall back to veth pair - Note: veth pair code is not merged yet).

So, if we choose "mode" option on CR to decide what type of network interface to use, we may need to have additional logic to handle Pod Scheduling (on a server with SRIOV capability) and a fallback option if there is an issue to get the VF assigned?.